purchases_ui_flutter/ios/Classes/PurchasesUiPaywallFooterView.swift

```diff
@@ -29,51 +34,76 @@ class PurchasesUiPaywallFooterViewFactory: NSObject, FlutterPlatformViewFactory
+class PurchasesUiPaywallFooterView: NSObject, FlutterPlatformView {
     private var _view: UIView
-    private let channel: FlutterMethodChannel?
+    private var _paywallProxy: PaywallProxy?
+    private var _methodChannel: FlutterMethodChannel
+    // Need to keep the controller in memory while this view is alive otherwise the delegate is dealocated
+    private var _paywallFooterViewController: PaywallFooterViewController
 
     init(
         frame: CGRect,
         viewIdentifier viewId: Int64,
         arguments args: Any?,
         binaryMessenger messenger: FlutterBinaryMessenger
     ) {
-        if #available(iOS 15.0, *) {
-            channel = FlutterMethodChannel(name: "purchases_ui_flutter/PaywallFooterView/\(viewId)",
-                                           binaryMessenger: messenger)
-            let paywallProxy = PaywallProxy()
-            let paywallFooterViewController = paywallProxy.createFooterPaywallView()
-            if let args = args as? [String: Any?] {
-                if let offeringId = args["offeringIdentifier"] as? String {
-                    paywallFooterViewController.update(with: offeringId)
-                }
+        _methodChannel = FlutterMethodChannel(name: "com.revenuecat.purchasesui/PaywallFooterView/\(viewId)",
+                                              binaryMessenger: messenger)
+        let paywallProxy = PaywallProxy()
+        _paywallProxy = paywallProxy
+        _paywallFooterViewController = paywallProxy.createFooterPaywallView()
+        if let args = args as? [String: Any?] {
+            if let offeringId = args["offeringIdentifier"] as? String {
+                _paywallFooterViewController.update(with: offeringId)
             }
-            guard let paywallFooterView = paywallFooterViewController.view else {
-                print("Error: error getting PaywallFooterView.")
-                _view = UIView()
-                super.init()
-                return
-            }
-            let newHeight = paywallFooterView.bounds.height
-            _view = paywallFooterView
-            channel?.invokeMethod("onHeightChanged", arguments: newHeight)
-        } else {
-            print("Error: attempted to present footer paywalls on unsupported iOS version.")
+        }
+        guard let paywallFooterView = _paywallFooterViewController.view else {
+            print("Error: error getting PaywallFooterView.")
             _view = UIView()
-            channel = nil
+            super.init()
+            return
         }
+        let newHeight = paywallFooterView.bounds.height
+        _view = paywallFooterView
+        _methodChannel.invokeMethod("onHeightChanged", arguments: newHeight)
         super.init()
+        _paywallProxy?.delegate = self;
     }
 
     func view() -> UIView {
         return _view
     }
+}
 
-    @available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
-    func paywallViewController(_ controller: PaywallViewController, didChangeSizeTo size: CGSize) {
```

Comment:
I hadn't actually setup the delegate in #999 so this was never called. After actually setting the delegate, this was returning a different value than the size as given by the view initially, which was making it draw incorrectly. I need to revisit whether this is needed and if so, why is it returning the wrong size...